pkg/azclient/auth_conf.go

```diff
@@ -46,6 +48,13 @@ type AzureAuthConfig struct {
 	// Auxiliary token provider for accessing resources from network tenant
 	// Require MSI to be enabled and have permission to access the KeyVault
 	AuxiliaryTokenProvider *AzureAuthAuxiliaryTokenProvider `json:"auxiliaryTokenProvider,omitempty" yaml:"auxiliaryTokenProvider,omitempty"`
+	// The path where a JSON file exists containing the JSON format of a UserAssignedIdentityCredentials struct
+	// See the msi-dataplane for more details on UserAssignedIdentityCredentials - https://github.com/Azure/msi-dataplane/blob/63fb37d3a1aaac130120624674df795d2e088083/pkg/dataplane/internal/generated_client.go#L156C6-L156C37
+	AADMSIDataPlaneIdentityPath string `json:"aadMSIDataPlaneIdentityPath,omitempty" yaml:"aadMSIDataPlaneIdentityPath,omitempty"`
```

Comment:
seems this is an internal data struct, are we going to consume it or also need to config the contents here?

And if need to config it, could we generate the contents based on our AzureAuthConfig and so we don't need to duplicate the same contents in two different files?

Reply:
> seems this is an internal data struct, are we going to consume it or also need to config the contents here?

Yeah it is an internal data struct, which is why this authentication type is only available to 1st party Microsoft applications. Cloud Provider would just consume the filepath where the JSON file lives and `dataplane.NewUserAssignedIdentityCredential` handles everything from there.